src/libraries/DrawAccumulatorLib.sol

```diff
@@ -273,12 +294,25 @@ library DrawAccumulatorLib {
         );
     }
 
+    /// @notice Computes the interim value C for the EWA
+    /// @param _alpha The EWA smoothing parameter.
+    /// @param _x The x value to compute for
+    /// @param _k The total available balance
+    /// @return The value C
     function computeC(SD59x18 _alpha, uint _x, uint _k) internal pure returns (SD59x18) {
         return toSD59x18(int(_k)).mul(_alpha.pow(toSD59x18(int256(_x))));
     }
 
-    /**
-     */
+    /// @notice Binary searches an array of draw ids for the given target draw id.
+    /// @param _drawRingBuffer The array of draw ids to search
+    /// @param _oldestIndex The oldest index in the ring buffer
+    /// @param _newestIndex The newest index in the ring buffer
+    /// @param _cardinality The number of items in the ring buffer
+    /// @param _targetLastCompletedDrawId The target draw id to search for
+    /// @return beforeOrAtIndex The index of the observation occurring at or before the target draw id
+    /// @return beforeOrAtDrawId The draw id of the observation occurring at or before the target draw id
+    /// @return afterOrAtIndex The index of the observation occurring at or after the target draw id
+    /// @return afterOrAtDrawId The draw id of the observation occurring at or after the target draw id 
     function binarySearch(
```

Comment:
In V4, we had the ObservationLib that was exporting this function: https://github.com/pooltogether/v4-core/blob/435e91a29dd7cd1553e6147cf33e9fa4ce02ef99/contracts/libraries/ObservationLib.sol#L15-L16
Maybe we should turn it into a package like we did for the RingBufferLib.

Reply:
That's possible.  I'll make a ticket for it.